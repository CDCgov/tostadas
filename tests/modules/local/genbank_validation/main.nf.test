nextflow_process {

    name "Test Process GENBANK_VALIDATION"
    script "modules/local/genbank_validation/main.nf"
    process "GENBANK_VALIDATION"

    test("Should clean and validate the FASTA file") {

        when {
            params {
            }

            process {
                """
                // Inputs
                input[0] = [ [ sample_id: "MZ568967" ], file("${projectDir}/assets/sample_fastas/rsv/MZ568967.fasta") ]
                """
            }
        }

        then {
            assert process.success

            // Extract output tuple
            def (meta, cleaned_fasta) = process.out[0]

            // Snapshot test
            assert snapshot(cleaned_fasta).match()
        }
    }
}
