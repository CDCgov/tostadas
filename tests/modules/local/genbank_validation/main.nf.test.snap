{
    "Should clean and validate the FASTA file": {
        "content": null,
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-08-28T16:21:46.340077568"
    }
}